### ISSUE TITLE
`stats.funcs._scipy_kraft_burrows_nousek` raises DeprecationWarning under Python 3.9

### ISSUE THREAD

**Issue by dhomeier** [astropy/astropy#10832]
### Description

As of Python 3.9.0 `test_scipy_poisson_limit` fails with a warning
```
>       assert_allclose(funcs._scipy_kraft_burrows_nousek(5., 2.5, .99),
                        (0, 10.67), rtol=1e-3)
...
>       factorial_N = float(math.factorial(N))
E       DeprecationWarning: Using factorial() with floats is deprecated
```
due to the function calling `math.factorial` on a `float` input:
https://github.com/astropy/astropy/blob/89580eee52dbe06e54fd51abe69ba0b0ecefc33a/astropy/stats/funcs.py#L1120

### Expected behavior

The function actually documents the input as `N : int or np.int32/np.int64`, but should check this or convert using `round`.

### Actual behavior
See above.

### Steps to Reproduce
```python
import astropy
astropy.test(package='stats', args='-vk scipy_poisson')
```

### System Details
macOS-10.14.6-x86_64-i386-64bit
Python 3.9.0 (default, Oct 10 2020, 10:27:03) 
[Clang 11.0.0 (clang-1100.0.33.17)]
Numpy 1.19.2
astropy 4.1rc2
astropy 4.2.dev863+ge114134a2
Scipy 1.5.2
Matplotlib 3.3.2

**Comment by pllim** [comment]
That's weird. How did the CI not catch this? We test against Python 3.9-dev in cron job.

https://github.com/astropy/astropy/blob/3df2b32810212d97aa754adf0a9cc08c5300e0ee/.travis.yml#L184

**Comment by bsipocz** [comment]
What does the python4 label mean here? Afaik there is no python4 on the roadmap, or are we using it as a meta label for dev python versions?

**Comment by pllim** [comment]
Re: Python4 label -- Not sure. There is no record on when it was created or who created it. 🤷

**Comment by bsipocz** [comment]
I recall it was created when we still have if/else for python2 vs 3 resulting code that would not run if there was a python4...

**Comment by pllim** [comment]
Hah... I think that boat has sailed...

**Comment by dhomeier** [comment]
I was just using it as a proxy for > 3.8...

**Comment by dhomeier** [comment]
Closed by #10838, thanks @pllim!
